src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,191 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    class Node<T> {
+        private T element;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(T element) {
+            this.element = element;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (first == null) {
+            addToEmptyList(value);
+        } else {
+            addLastNode(value);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndex(index, size + 1);
```

Comment:
1. You shouldn't allow add with index = size + 1 as your actual last index is size - 1, so size is limit for add operation:
```suggestion
        checkIndex(index, size);
```

2. After such change all your checkIndex method will be called with (index, size) arguments so size may be omitted from method totally:
```suggestion
        checkIndex(index);
```